backend/Controllers/WhaleController.cs

```diff
@@ -22,5 +22,12 @@ public ActionResult<ListResponse<Species>> GetAllSpecies()
             var whales = _whales.GetAllSpecies();
             return new ListResponse<Species>(whales);
         }
+        
+        [HttpGet("{speciesId}")]
+        public ActionResult<Species> GetSpeciesById([FromRoute] int speciesId)
+        {
+            var whale = _whales.GetSpeciesById(speciesId);
+            return whale;
```

Comment:
Good, but please add some error handling if the ID isn't valid - the `Single()` down in the Repo will throw an `InvalidOperationException` ([see here](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.single?view=net-7.0#system-linq-enumerable-single-1(system-collections-generic-ienumerable((-0))))) if it can't find a species with that ID, so you can catch it and return a 404 Not Found:
```suggestion
            try
            {
                var whale = _whales.GetSpeciesById(speciesId);
                return whale;
            }
            catch (InvalidOperationException)
            {
                return NotFound();
            }
```

Reply:
done, thanks